pymatgen/core/structure.py

```diff
@@ -3980,67 +3984,174 @@ def set_charge(self, new_charge: float = 0.0) -> None:
         steps: int = 500,
         fmax: float = 0.1,
+        opt_kwargs: dict = None,
         return_trajectory: bool = False,
         verbose: bool = False,
-    ) -> Structure | tuple[Structure, TrajectoryObserver]:
+    ) -> Structure | tuple[Structure, Trajectory]:
         """
-        Performs a crystal structure relaxation using some algorithm.
+        Performs a crystal structure relaxation using an ASE calculator.
 
         Args:
-            calculator: A string or an ASE calculator. Defaults to 'm3gnet', i.e. the M3GNet universal potential.
+            calculator: An ASE Calculator or a string from the following options: "m3gnet.
+                Defaults to 'm3gnet', i.e. the M3GNet universal potential.
             relax_cell (bool): whether to relax the lattice cell. Defaults to True.
+            optimizer (str): name of the ASE optimizer class to use
             stress_weight (float): the stress weight for relaxation. Defaults to 0.01.
             steps (int): max number of steps for relaxation. Defaults to 500.
             fmax (float): total force tolerance for relaxation convergence.
                 Here fmax is a sum of force and stress forces. Defaults to 0.1.
+            opt_kwargs (dict): kwargs for the ASE optimizer class.
             return_trajectory (bool): Whether to return the trajectory of relaxation.
                 Defaults to False.
             verbose (bool): whether to print out relaxation steps. Defaults to False.
 
         Returns:
-            Structure: IAP-relaxed structure
+            Structure: Relaxed structure
         """
         import contextlib
         import io
         import sys
 
         from ase.constraints import ExpCellFilter
-        from ase.optimize.fire import FIRE
-        from m3gnet.models import M3GNet, M3GNetCalculator, Potential
-        from m3gnet.models._dynamics import TrajectoryObserver
+        from ase.io import read
+        from ase.optimize import BFGS, FIRE, LBFGS, BFGSLineSearch, GPMin, LBFGSLineSearch, MDMin
 
         from pymatgen.io.ase import AseAtomsAdaptor
 
-        if calculator == "m3gnet":
-            potential = Potential(M3GNet.load())
-            calculator = M3GNetCalculator(potential=potential, stress_weight=stress_weight)
+        opt_kwargs = opt_kwargs or {}
+
+        # Get optimizer
+        if optimizer.lower() == "bfgs":
+            opt_class = BFGS
+        elif optimizer.lower() == "bfgslinesearch":
+            opt_class = BFGSLineSearch
+        elif optimizer.lower() == "lbfgs":
+            opt_class = LBFGS
+        elif optimizer.lower() == "lbfgslinesearch":
+            opt_class = LBFGSLineSearch
+        elif optimizer.lower() == "gpmin":
+            opt_class = GPMin
+        elif optimizer.lower() == "mdmin":
+            opt_class = MDMin
+        elif optimizer.lower() == "fire":
+            opt_class = FIRE
+        else:
+            raise ValueError(f"Unknown optimizer: {optimizer}")
```

Comment:
Didn't test this but I would prob enforce users use the right casing in which case you could simplify to:

```suggestion
        from ase import optimize
        try:
            opt_class = getattr(optimize, optimizer)
        except AttributeError:
            raise ValueError(f"Unknown {optimizer=}, must be one of {list(dir(optimize))}")
```

Reply:
Brilliant. I knew there was a cleaner way to do this but didn't know how!